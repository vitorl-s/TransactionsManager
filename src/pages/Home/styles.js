import {StyleSheet, Platform} from 'react-native';
import {Colors} from '../../constants/Colors';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 10,
    backgroundColor: Colors.Background,
  },
  balance: {
    justifyContent: 'center',
    width: '85%',
    alignItems: 'center',
    alignSelf: 'center',
    padding: 5,
    paddingBottom: 15,
    marginTop: 15,
    borderRadius: 25,
    backgroundColor: 'white',
  },
  title: {
    alignSelf: 'center',
    color: Colors.Text,
    fontSize: 22,
    fontWeight: 'bold',
  },
  FAB: {
    position: 'absolute',
    margin: 16,
    right: 0,
    backgroundColor: Colors.ButtonColor,
    bottom: 0,
  },
  noTransactionsContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  noTransactions: {
    color: 'white',
    fontSize: 16,
    fontWeight: 'bold',
  },
  listContainer: {
    paddingTop: 20,
    paddingBottom: 20,
  },
});

export default styles;
