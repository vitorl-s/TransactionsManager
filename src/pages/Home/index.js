import React, {useState, useEffect} from 'react';
import {FAB} from 'react-native-paper';
import {View, Text} from 'react-native';
import {useSelector} from 'react-redux';
import {useNavigation} from '@react-navigation/native';
import styles from './styles';
import Transactions from '../../components/Transactions';

const Home = () => {
  const navigation = useNavigation();
  const transactionList = useSelector((item) => item.transactions.transactions);
  const [balance, setBalance] = useState(0);

  useEffect(() => {
    calculateBalance();
  }, [transactionList]);

  const calculateBalance = () => {
    let finalBalance = transactionList.reduce((a, b) => a + b.value, 0);
    setBalance(finalBalance);
  };

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Transações</Text>
      <View style={styles.balance}>
        <Text style={styles.balanceText}>
          Saldo: R$ {balance.toFixed(2).toString().replace('.', ',')}
        </Text>
      </View>
      {transactionList.length > 0 ? (
        <View style={styles.listContainer}>
          <Transactions />
        </View>
      ) : (
        <View style={styles.noTransactionsContainer}>
          <Text style={styles.noTransactions}>
            Você ainda não possui transações cadastradas
          </Text>
        </View>
      )}
      <FAB
        style={styles.FAB}
        icon="plus"
        onPress={() => navigation.navigate('AddTransaction')}
      />
    </View>
  );
};

export default Home;
