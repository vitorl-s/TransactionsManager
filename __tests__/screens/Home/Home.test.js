import React from 'react';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import Home from '../../../src/pages/Home';
import {render} from '@testing-library/react-native';
import MockedNavigator from '../../../src/navigator/mockedNavigator';

test('should render HomePage with no transactions message', () => {
  jest.mock('react-redux', () => ({
    useSelector: jest
      .fn()
      .mockImplementation((selector) => selector.transactions.transactions),
  }));
  const initialState = {
    transactions: {
      transactions: [],
    },
  };
  const mockStore = configureStore();
  const store = mockStore(initialState);
  const {toJSON} = render(
    <Provider store={store}>
      <MockedNavigator component={Home} />
    </Provider>,
  );
  expect(toJSON()).toMatchSnapshot();
});

test('should render HomePage with transactions', () => {
  jest.mock('react-redux', () => ({
    useSelector: jest
      .fn()
      .mockImplementation((selector) => selector.transactions.transactions),
  }));
  const initialState = {
    transactions: {
      transactions: [{value: 12.5, desc: 'testing desc', date: '09/08/2020'}],
    },
  };
  const mockStore = configureStore();
  const store = mockStore(initialState);
  const {toJSON} = render(
    <Provider store={store}>
      <MockedNavigator component={Home} />
    </Provider>,
  );
  expect(toJSON()).toMatchSnapshot();
});
