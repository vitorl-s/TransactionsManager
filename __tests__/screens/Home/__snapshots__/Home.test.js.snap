// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HomePage with no transactions message 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "zIndex": 1,
          },
          false,
        ]
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                MockedScreen
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "#1E1F28",
                          "flex": 1,
                          "paddingTop": 10,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "alignSelf": "center",
                            "color": "#F5F5F5",
                            "fontSize": 22,
                            "fontWeight": "bold",
                          }
                        }
                      >
                        Transações
                      </Text>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "justifyContent": "center",
                            "marginTop": 15,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#F5F5F5",
                              "fontSize": 30,
                              "fontWeight": "bold",
                            }
                          }
                        >
                          Saldo: R$ 
                          0,00
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "justifyContent": "center",
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#F5F5F5",
                              "fontSize": 16,
                              "fontWeight": "bold",
                            }
                          }
                        >
                          Você ainda não possui transações cadastradas
                        </Text>
                      </View>
                      <View
                        pointerEvents="auto"
                        style={
                          Object {
                            "backgroundColor": "#EF3651",
                            "borderRadius": 28,
                            "bottom": 0,
                            "elevation": 6,
                            "margin": 16,
                            "opacity": 1,
                            "position": "absolute",
                            "right": 0,
                            "shadowColor": "#000000",
                            "shadowOffset": Object {
                              "height": 5,
                              "width": 0,
                            },
                            "shadowOpacity": 0.24,
                            "shadowRadius": 6,
                            "transform": Array [
                              Object {
                                "scale": 1,
                              },
                            ],
                          }
                        }
                      >
                        <View
                          accessibilityRole="button"
                          accessibilityState={
                            Object {
                              "disabled": undefined,
                            }
                          }
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Array [
                              Object {
                                "overflow": "hidden",
                              },
                              Object {
                                "borderRadius": 28,
                              },
                            ]
                          }
                        >
                          <View
                            pointerEvents="none"
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "flexDirection": "row",
                                  "justifyContent": "center",
                                },
                                Object {
                                  "height": 56,
                                  "width": 56,
                                },
                              ]
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "justifyContent": "center",
                                  },
                                  Object {
                                    "height": 24,
                                    "width": 24,
                                  },
                                ]
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "bottom": 0,
                                    "left": 0,
                                    "opacity": 1,
                                    "position": "absolute",
                                    "right": 0,
                                    "top": 0,
                                    "transform": Array [
                                      Object {
                                        "rotate": "0deg",
                                      },
                                    ],
                                  }
                                }
                              >
                                <Text
                                  accessibilityElementsHidden={true}
                                  allowFontScaling={false}
                                  importantForAccessibility="no-hide-descendants"
                                  pointerEvents="none"
                                  style={
                                    Array [
                                      Object {
                                        "color": "#ffffff",
                                        "fontSize": 24,
                                      },
                                      Array [
                                        Object {
                                          "transform": Array [
                                            Object {
                                              "scaleX": 1,
                                            },
                                          ],
                                        },
                                        Object {
                                          "backgroundColor": "transparent",
                                        },
                                      ],
                                      Object {
                                        "fontFamily": "Material Design Icons",
                                        "fontStyle": "normal",
                                        "fontWeight": "normal",
                                      },
                                      Object {},
                                    ]
                                  }
                                >
                                  󰐕
                                </Text>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;

exports[`should render HomePage with transactions 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "zIndex": 1,
          },
          false,
        ]
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                MockedScreen
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "#1E1F28",
                          "flex": 1,
                          "paddingTop": 10,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "alignSelf": "center",
                            "color": "#F5F5F5",
                            "fontSize": 22,
                            "fontWeight": "bold",
                          }
                        }
                      >
                        Transações
                      </Text>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "justifyContent": "center",
                            "marginTop": 15,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#F5F5F5",
                              "fontSize": 30,
                              "fontWeight": "bold",
                            }
                          }
                        >
                          Saldo: R$ 
                          12,50
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "paddingBottom": 20,
                            "paddingTop": 20,
                          }
                        }
                      >
                        <RCTScrollView
                          contentContainerStyle={
                            Object {
                              "paddingBottom": 80,
                              "paddingTop": 20,
                            }
                          }
                          data={
                            Array [
                              Object {
                                "date": "09/08/2020",
                                "desc": "testing desc",
                                "value": 12.5,
                              },
                            ]
                          }
                          disableVirtualization={false}
                          extraData={
                            Array [
                              Object {
                                "date": "09/08/2020",
                                "desc": "testing desc",
                                "value": 12.5,
                              },
                            ]
                          }
                          getItem={[Function]}
                          getItemCount={[Function]}
                          horizontal={false}
                          initialNumToRender={10}
                          keyExtractor={[Function]}
                          maxToRenderPerBatch={10}
                          onContentSizeChange={[Function]}
                          onEndReachedThreshold={2}
                          onLayout={[Function]}
                          onMomentumScrollEnd={[Function]}
                          onScroll={[Function]}
                          onScrollBeginDrag={[Function]}
                          onScrollEndDrag={[Function]}
                          removeClippedSubviews={false}
                          renderItem={[Function]}
                          scrollEventThrottle={50}
                          stickyHeaderIndices={Array []}
                          updateCellsBatchingPeriod={50}
                          viewabilityConfigCallbackPairs={Array []}
                          windowSize={21}
                        >
                          <View>
                            <View
                              onLayout={[Function]}
                              style={null}
                            >
                              <View
                                style={
                                  Object {
                                    "alignSelf": "center",
                                    "backgroundColor": "#2A2C36",
                                    "borderRadius": 25,
                                    "height": 85,
                                    "justifyContent": "center",
                                    "marginTop": 15,
                                    "padding": 5,
                                    "paddingBottom": 15,
                                    "width": "85%",
                                  }
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "alignItems": "center",
                                      "flexDirection": "row",
                                      "justifyContent": "space-between",
                                      "marginHorizontal": 20,
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "flex": 0.6,
                                        "marginRight": 10,
                                      }
                                    }
                                  >
                                    <Text
                                      style={
                                        Object {
                                          "color": "green",
                                          "fontSize": 16,
                                          "fontWeight": "bold",
                                          "marginTop": 10,
                                        }
                                      }
                                    >
                                      R$ 
                                      12.5
                                    </Text>
                                    <Text
                                      style={
                                        Object {
                                          "color": "#ABB4BD",
                                          "fontSize": 12,
                                        }
                                      }
                                    >
                                      09/08/2020
                                    </Text>
                                  </View>
                                  <View
                                    style={
                                      Object {
                                        "flex": 1,
                                      }
                                    }
                                  >
                                    <Text
                                      numberOfLines={4}
                                      style={
                                        Object {
                                          "color": "#F5F5F5",
                                          "marginTop": 10,
                                        }
                                      }
                                    >
                                      testing desc
                                    </Text>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </View>
                        </RCTScrollView>
                      </View>
                      <View
                        pointerEvents="auto"
                        style={
                          Object {
                            "backgroundColor": "#EF3651",
                            "borderRadius": 28,
                            "bottom": 0,
                            "elevation": 6,
                            "margin": 16,
                            "opacity": 1,
                            "position": "absolute",
                            "right": 0,
                            "shadowColor": "#000000",
                            "shadowOffset": Object {
                              "height": 5,
                              "width": 0,
                            },
                            "shadowOpacity": 0.24,
                            "shadowRadius": 6,
                            "transform": Array [
                              Object {
                                "scale": 1,
                              },
                            ],
                          }
                        }
                      >
                        <View
                          accessibilityRole="button"
                          accessibilityState={
                            Object {
                              "disabled": undefined,
                            }
                          }
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Array [
                              Object {
                                "overflow": "hidden",
                              },
                              Object {
                                "borderRadius": 28,
                              },
                            ]
                          }
                        >
                          <View
                            pointerEvents="none"
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "flexDirection": "row",
                                  "justifyContent": "center",
                                },
                                Object {
                                  "height": 56,
                                  "width": 56,
                                },
                              ]
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "justifyContent": "center",
                                  },
                                  Object {
                                    "height": 24,
                                    "width": 24,
                                  },
                                ]
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "bottom": 0,
                                    "left": 0,
                                    "opacity": 1,
                                    "position": "absolute",
                                    "right": 0,
                                    "top": 0,
                                    "transform": Array [
                                      Object {
                                        "rotate": "0deg",
                                      },
                                    ],
                                  }
                                }
                              >
                                <Text
                                  accessibilityElementsHidden={true}
                                  allowFontScaling={false}
                                  importantForAccessibility="no-hide-descendants"
                                  pointerEvents="none"
                                  style={
                                    Array [
                                      Object {
                                        "color": "#ffffff",
                                        "fontSize": 24,
                                      },
                                      Array [
                                        Object {
                                          "transform": Array [
                                            Object {
                                              "scaleX": 1,
                                            },
                                          ],
                                        },
                                        Object {
                                          "backgroundColor": "transparent",
                                        },
                                      ],
                                      Object {
                                        "fontFamily": "Material Design Icons",
                                        "fontStyle": "normal",
                                        "fontWeight": "normal",
                                      },
                                      Object {},
                                    ]
                                  }
                                >
                                  󰐕
                                </Text>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
